### DISCUSSION TITLE
Is "CSS Only" Deprecated?

### DISCUSSION THREAD

**Discussion by benash** [vmware/clarity#5736] [Q&A]
I see it [mentioned on GitHub](https://github.com/vmware/clarity/blob/next/docs/INSTALLATION.md#installing-clarity-ui-css-only) that it's possible to use Clarity by importing `@clr/ui/clr-ui.css` and simply using whatever class attributes are needed (the traditional sort of way). I've found it necessary to do this for some layout components that are not in the React package. But I also find it to be a really valuable tool to fall back on, in situations where we'd rather design the component interface ourselves. So I'm wondering: will this CSS-only approach be supported going forward, or is the project moving away from this in favor of only encapsulated styles?

**Comment by gnomeontherun** [comment]
Hi @benash. The @clr/ui package will continue to exist for as long as you need it. As we build out the Clarity Core (with React support) packages the need for @clr/ui will eventually disappear as everything will be encapsulated into the Core components. If you wish to design the components yourself, then you may find value in using the new Clarity Core components as they are released or borrowing the pertinent code from @clr/ui to eventually remove the dependency.

It might not be necessary to carry over the @clr/ui styles even if you want to build React components based on those original styles, as you might be better off utilizing the design tokens for this. Part of what we're doing as we rebuilt everything using Clarity Core is reassess some of the foundational aspects of the components for better design, customization, and accessibility support. The more you depend upon our officially supported APIs the better support you'll get long term from using Clarity.

**Reply by benash** [reply]
OK, I think I understand—it sounds like the @clr/ui package will continue to be available as a way of referencing the underlying CSS (which is likely to change over time), whereas the Core packages provide a stable interface via components. That sounds great (if I'm understanding correctly)—thanks!
